Addition of the GC modules to the user manual (#32)
haddocking/haddock3-user-manual: src/modules/topology.md

Reviewer: ```suggestion
- [`[topoaa]` module](#topoaa-module)
- [`[topocg]` module](#topocg-module)

```

# Topology modules

- [`[topoaa]` module](#topoaa-module)
- [`[topoaa]` module](#topoaa-module)

> - [`[topocg]` module](#topocg-module)

Reviewer: ```suggestion
- [`[topoaa]` module](#topoaa-module)
- [`[topocg]` module](#topocg-module)
```

  - `5_phosphate`: Allows to define the state of the 5' end of nucleic acids sequences. If set to `true`, 5' end will be a phosphate group. Otherwise it will be an OH. (default false). Note that chain breaks are not evaluated as 5' ends.

<hr>

## `[topocg]` module

> The `[topocg]` module is dedicated to the mapping of the all-atom (AA) structures generated by the `[topoaa]` into their coarse-grained (CG) MARTINI2 representations and topologies. The associated CNS compatible topologies (.psf) are built for each of the input structures, as well as the necessary distance restraint tables for backmapping (_cg_to_aa.tbl).

Reviewer: ```suggestion
The `[topocg]` module is dedicated to the mapping of the all-atom (AA) structures generated by the `[topoaa]` into their coarse-grained (CG) MARTINI2 representations and topologies. The associated CNS compatible topologies (`.psf`) are built for each of the input structures, as well as the necessary distance restraint tables for backmapping (`*_cg_to_aa.tbl`).
```

It will:
- Build the CG structure from the AA, based on the MARTINI mapping (`.pdb`)
- Build and write out topologies (`.psf`)
- Build and write out distance restraint files (`_cg_to_aa.tbl`) necessary for mapping the CG structures back to AA.

> The restraints created define atom-to-bead restraints, with one restraint per bead set to 0Å between the bead and the geometric center of its corresponding atoms (see the [cgtoaa] module for further details).

Reviewer: ```suggestion
The restraints created define atom-to-bead restraints, with one restraint per bead set to 0Å between the bead and the geometric center of its corresponding atoms (see the [`[cgtoaa]`](https://www.bonvinlab.org/haddock3-user-manual/modules/refinement.html#cgtoaa-module) module for further details).
```

This module is a prerequisite when running CG docking.
The topologies built are necessary for the subsequent sampling and refinement modules.

> Therefore, in a CG docking pipeline, it is necessary to run this module after the `[topoaa]` module (second step of the workflow).

Reviewer: to some extend, this needs to be refactored, but let's worry about that later I would say

Author: what do you mean ? You would like to not have the topoaa module before the topocg ?

Reviewer: In a CG workflow, you could start with `[topocg]` and if you miss the all atom topology for the backmapping, you can build it within the `[topocg]`  module.
But no need to worry about that now

> Therefore, in a CG docking pipeline, it is necessary to run this module after the `[topoaa]` module (second step of the workflow).

Reviewer: ```suggestion
**Note** that, with the current implementation, it is **necessary** to run this module right after the `[topoaa]` module.
```

> Note that currently the module does not support non-standard biomolecules (apart from standard amino-acids, DNA, RNA), such as small-molecules and coarbohydrates. Those are ignored in CG mode, but will re-appear if present in the initial AA structure after the back-mapping stage (`cgtoaa`) (provided parameters and topologies for the ligand have been provided or automatically generated).

Reviewer: ```suggestion
**Note** that currently, apart from standard amino-acids, DNA, RNA,  **the module does not support non-standard biomolecules**, such as small-molecules and carbohydrates. Those are ignored in CG mode, but will re-appear if present in the initial AA structure after the back-mapping stage (`[cgtoaa]` module) (provided parameters and topologies for the ligand have been provided or automatically generated).
```

> More information about `[topoaa]` parameters can be accessed [here](https://bonvinlab.org/haddock3/modules/topology/haddock.modules.topology.topocg.html#default-parameters) or retrieved by running:

Reviewer: ```suggestion
More information about `[topocg]` parameters can be accessed [here](https://bonvinlab.org/haddock3/modules/topology/haddock.modules.topology.topocg.html#default-parameters) or retrieved by running:
```

```bash
haddock3-cfg -m topocg
```

Here an example configuration file snapshot of a typical execution of the
`[topocg]` module:

```toml
# Definition of

> run_dir = "example"

Reviewer: ```suggestion
run_dir = "example_CG"
```

molecules = [
 "1abc.pdb",
 "2abc.pdb",
]

[topoaa]

[topocg]
cgffversion = "martini2"

# Workflow continues with other modules

> # ...
> ```
> 
> The most important parameter for the `[topocg]` module is:

Reviewer: ```suggestion

#### Notable parameters

The most important parameter for the `[topocg]` module is:
```

> - `cgffversion`: defines the force-field version for the CG mapping and parameters. Currently on MARTINI2 is supported.

Reviewer: ```suggestion
- `cgffversion`: defines the force-field version for the CG mapping and parameters. Currently only MARTINI2 is supported.
```
